common-pg/src/main/scala/com/socrata/pg/store/PGSecondaryDatasetMapWriter.scala

```diff
@@ -148,29 +148,72 @@ class PGSecondaryDatasetMapWriter[CT](override val conn: Connection,
     }
   }
 
-  def insertLocalRollupQuery = "INSERT INTO rollup_map (name, copy_system_id, soql, table_name) VALUES (?, ?, ?, ?)"
-  def updateLocalRollupQuery = "UPDATE rollup_map SET soql = ?, table_name = ? WHERE name = ? AND copy_system_id = ?"
+  def insertLocalRollupQuery = "INSERT INTO rollup_map (name, copy_system_id, soql, table_name) VALUES (?, ?, ?, ?) returning system_id"
+  def updateLocalRollupQuery = "UPDATE rollup_map SET soql = ?, table_name = ? WHERE name = ? AND copy_system_id = ? returning system_id"
   override def createOrUpdateRollup(copyInfo: CopyInfo, name: RollupName, soql: String, rawSoql: Option[String]): LocalRollupInfo = {
     val tableName = LocalRollupInfo.tableName(copyInfo, name)
+    var system_id = -1L
     rollup(copyInfo, name) match {
       case Some(_) =>
         using(conn.prepareStatement(updateLocalRollupQuery)) { stmt =>
           stmt.setString(1, soql)
           stmt.setString(2, tableName)
           stmt.setString(3, name.underlying)
           stmt.setLong(4, copyInfo.systemId.underlying)
-          t("create-or-update-rollup", "action" -> "update", "copy-id" -> copyInfo.systemId, "name" -> name)(stmt.executeUpdate())
+          t("create-or-update-rollup", "action" -> "update", "copy-id" -> copyInfo.systemId, "name" -> name)(
+            using(stmt.executeQuery()){resultSet =>
+              if(resultSet.next()){
```

Comment:
This (and the None branch) should be much more unhappy if it somehow fails to have a row returned.  That'd also let the `match` just return the systemId instead of setting it as a side-effect.